pkg/datagatherer/k8s/dynamic_test.go

```diff
@@ -130,98 +130,103 @@ func TestDynamicGatherer_Fetch(t *testing.T) {
-			),
-		},
-		"Secret resources should have data removed": {
-			gvr: schema.GroupVersionResource{Group: "", Version: "v1", Resource: "secrets"},
-			objects: []runtime.Object{
-				getSecret("testsecret", "testns1", map[string]interface{}{
-					"secretKey": "secretValue",
-				}, false, true),
-				getSecret("anothertestsecret", "testns2", map[string]interface{}{
-					"secretNumber": "12345",
-				}, false, true),
-			},
-			expected: asUnstructuredList(
-				getSecret("testsecret", "testns1", nil, false, false),
-				getSecret("anothertestsecret", "testns2", nil, false, false),
-			),
-		},
-		"Secret of type kubernetes.io/tls should have crts and not keys": {
-			gvr: schema.GroupVersionResource{Group: "", Version: "v1", Resource: "secrets"},
-			objects: []runtime.Object{
-				getSecret("testsecret", "testns1", map[string]interface{}{
-					"tls.key": "secretValue",
-					"tls.crt": "value",
-					"ca.crt":  "value",
-				}, true, true),
-				getSecret("anothertestsecret", "testns2", map[string]interface{}{
-					"example.key": "secretValue",
-					"example.crt": "value",
-				}, true, true),
-			},
-			expected: asUnstructuredList(
-				// only tls.crt and ca.cert remain
-				getSecret("testsecret", "testns1", map[string]interface{}{
-					"tls.crt": "value",
-					"ca.crt":  "value",
-				}, true, false),
-				// all other keys removed
-				getSecret("anothertestsecret", "testns2", nil, true, false),
-			),
-		},
-		"Foos in different namespaces should be returned if they are in the namespace list for the gatherer": {
-			gvr:        schema.GroupVersionResource{Group: "foobar", Version: "v1", Resource: "foos"},
-			namespaces: []string{"testns", "testns2"},
-			objects: []runtime.Object{
-				getObject("foobar/v1", "Foo", "testfoo", "testns", false),
-				getObject("foobar/v1", "Foo", "testfoo2", "testns2", false),
-				getObject("foobar/v1", "Foo", "testfoo3", "nottestns", false),
-			},
-			expected: asUnstructuredList(
-				getObject("foobar/v1", "Foo", "testfoo", "testns", false),
-				getObject("foobar/v1", "Foo", "testfoo2", "testns2", false),
-			),
-		},
-		"Resources should have managed fields removed": {
-			gvr: schema.GroupVersionResource{Group: "apps", Version: "v1", Resource: "deployments"},
-			objects: []runtime.Object{
-				getObject("apps/v1", "Deployment", "foo1", "testns", false),
-				getObject("apps/v1", "Deployment", "foo2", "testns", true),
-			},
-			expected: asUnstructuredList(
-				getObject("apps/v1", "Deployment", "foo1", "testns", false),
-				getObject("apps/v1", "Deployment", "foo2", "testns", false),
-			),
-		},
-		// Note that we can't test use of fieldSelector to exclude namespaces
-		// here as the as the fake client does not implement it.
-		// See go/pkg/mod/k8s.io/client-go@v0.17.0/dynamic/fake/simple.go:291
-		// TODO: Add a custom reactor to allow testing of fieldSelector.
+		// TODO restore these test cases by adding to the new list of gvr to list type mappings, all mapp to UnstructuredList
```

Comment:
@wwwil I think I've worked out what's needed to fix this if we do go this route, see below.